### ISSUE TITLE
Emitting diffs in addition to the full row.

### ISSUE THREAD

**Issue by joshcanfield** [zendesk/maxwell#144]
It would be useful to support a configuration that emits a diff record, only including the columns that have changed.

The old column data appears to be available in update events. I'm interested to know if this is something that has been considered? And if so is there a roadmap or design?

**Comment by osheroff** [comment]
yup, this is on the list, and is actually a fairly simply change.  I had figured it would look like:

```
  { 
     "type": "update", 
     "data": { "id": 1, "txt": "abc", "foo": "bar"  },  // the complete row image 
     "old": { "txt": "old_text" } // only the rows that changed
  }
```

that way parsing an update stays the same as parsing an insert (`data` has the full row image), but you can access what the row was if you like. 

I don't have a particular time frame on implementing this, of course I'm super patches welcome, just ping this issue if you start on code so I don't duplicate effort.

**Comment by osheroff** [comment]
I should probably pull together a ROADMAP.md at some point too.

**Comment by joshcanfield** [comment]
This is great to hear, and thanks for being responsive!

I have a hackday tomorrow and I'm going to be integrating Maxwell into our system and I need column level change notification. If someone isn't already implementing this feature then I'll spend some time creating a branch.

**Comment by osheroff** [comment]
great to hear, josh.  No one's working on it at the moment as far as I know.

**Comment by jpeyper** [comment]
This is something I would be super keen for. 

Has anyone looked at this yet?

**Comment by joshcanfield** [comment]
I wasn't able to get to it, but kristiankaufmann pull request looks good! :+1:

**Comment by osheroff** [comment]
Fixed in 0.16.0
